### ISSUE TITLE
tags vs extraTags vs context are confusing

### ISSUE THREAD

**Issue by mikebell90** [getsentry/sentry-java#443]
According to the docs there are tags and extratags, normally configured in sentry.properties.

1. What's the exact and specific different between these.
2. Let's say I want to have the following tags show up in sentry, all searchable

user (or some such principal)
thread
requestId

Now, are any of these handled automatically by the servlet listener? If not what do I need to do. I can clearly add them to the MDC in a filter or elsewhere. But do I also have to statically define them in sentry.properties as tags, extratags,or what?

**Comment by bretthoerner** [comment]
Tags configured via the options will be static for the life of the program. You can also add tags in the `EventBuilder` if you are manually sending events, or in an `EventBuilderHelper` you add to your client (from memory, roughly: `Sentry.getStoredClient().addBuilderHelper(ebh);`)

Extra tags exist to pull data from the MDC (and should probably be renamed) out into a (dynamic) tag, so if you set extratags of `foo,bar` it should look for `foo` and `bar` in the MDC.

To make it slightly more confusing, `user` is a first class thing in Sentry, and you can set it in your context via `Sentry.setUser(UserBuilder...);`. Now that I think of it, we should allow setting tags in the current context that way, too. I'll add this next week.

So in short, for now, it sounds like you want extratags + `Sentry.setUser`. Hope that helps.

**Comment by mikebell90** [comment]
Thanks, I'll look into that. It does clarify things. Will leave open for a few minutes while I ponder the userBuilder, should be closed shortly
